core/impl/src/main/java/com/blazebit/persistence/impl/AbstractCTECriteriaBuilder.java

```diff
@@ -201,18 +201,16 @@ public CTEInfo createCTEInfo() {
     }
 
     protected List<String> prepareAndGetAttributes() {
-        List<String> attributes = new ArrayList<String>(bindingMap.size());
         for (Map.Entry<String, Integer> bindingEntry : bindingMap.entrySet()) {
             final String attributeName = bindingEntry.getKey();
 
             ExtendedAttribute attributeEntry = attributeEntries.get(attributeName);
             List<Attribute<?, ?>> attributePath = attributeEntry.getAttributePath();
-            attributes.add(attributeName);
+            JpaMetamodelAccessor jpaMetamodelAccessor = mainQuery.jpaProvider.getJpaMetamodelAccessor();
 
-            if (JpaMetamodelUtils.isJoinable(attributePath.get(attributePath.size() - 1))) {
-                // We have to map *-to-one relationships to their ids
-                EntityType<?> type = mainQuery.metamodel.entity(attributeEntry.getElementClass());
-                Attribute<?, ?> idAttribute = JpaMetamodelUtils.getSingleIdAttribute(type);
+            if (jpaMetamodelAccessor.isJoinable(attributePath.get(attributePath.size() - 1))) {
```

Comment:
Now that we support collections in CTE's (do we? We dropped a few earlier constraints here) We may also want to include `!isCollection` here.

Reply:
The thing is, you can't bind collections that are contained in CTEs, so this shouldn't be a problem here. IMO we should throw an exception during bind when encountering a plural attribute and not worry about this here.